src/dune_engine/sandbox_mode.ml

```diff
@@ -70,51 +78,43 @@ module Dict = struct
 end
 
 module Set = struct
-  type key = t
-  type t = bool Dict.t
-
-  let compare = Dict.compare Bool.compare
-  let of_func = Dict.of_func
-  let singleton k = of_func (equal k)
+  module T = struct
+    type nonrec t = t
+
+    let to_int = function
+      | None -> 1
```

Comment:
Can we `Make` a `Bit_set` for `t option` as well? (Just thinking about how we can move as much code as possible into `Bit_set`.)

Reply:
Do you mean make a functor like `Bit_set.Make` that works for `t option` rather `t`?

Reply:
I thought that maybe there should be two functors: one for `t` and another one for `t option`. But maybe I'm overcomplicating things. Feel free to ignore this comment if you'd rather keep the code as is.

Reply:
It can work, with a signature that looks something like this:
```
module type Element = sig
  type t

  (** [to_int t] must return a unique number for every [t] between [1] and
      [Sys.int_size - 1] (inclusive) *)
  val to_int : t -> int

  (** [all] contains all possible set elements *)
  val all : t list

  val to_dyn : t -> Dyn.t
end

module Option (T : Element) : Element with type t = T.t option

module Make (Element : Element) : sig ...
end
```

But it seems a bit overkill if you aren't using this anywhere else yet.

Reply:
OK, agreed, let's stick to the existing `Make`.